pkg/store/employee/employee.go

```diff
@@ -33,6 +54,82 @@ func (s *store) All() ([]*model.Employee, error) {
 	return employees, s.db.Find(&employees).Error
 }
 
+func (s *store) Save(employee *model.Employee) error {
+	return s.db.Save(&employee).Error
+}
+
+func (s *store) Update(body EditEmployeeRequest, id string) (*model.Employee, error) {
+
+	var employee *model.Employee
+
+	return employee, s.db.Transaction(func(tx *gorm.DB) error {
+
+		// 1 update employee table
+
+		// 1.1 get employee by employee id
+		err := tx.Where("id = ?", id).First(&employee).Error
+		if err != nil {
+			return err
+		}
+
+		// 1.2 update infor
+		employee.FullName = body.Fullname
+		employee.TeamEmail = body.Email
+		employee.PhoneNumber = body.Phone
+		employee.LineManagerID = body.LineManager
+		employee.DiscordID = body.DiscordID
+		employee.GithubID = body.GithubID
+		employee.ChapterID = body.Chapter
+		employee.SeniorityID = body.Seniority
+		employee.DateOfBirth = body.DoB
+		employee.Gender = body.Gender
+		employee.Address = body.Address
+		employee.PersonalEmail = body.PersonalEmail
+
+		// 1.3 save to DB
+		err = tx.Table("employees").Updates(&employee).Error
+		if err != nil {
+			return err
+		}
+
+		// 2 update role for employee
+
+		// 2.1 delete all roles of the employee
+		employeeRole := model.EmployeeRoles{}
+		tx.Table("employee_roles").Where("employee_id = ?", id).Delete(&employeeRole)
+
+		// 2.2 create role for employee
+		employeeRole.ID = model.NewUUID()
```

Comment:
`ID` will auto create by postgres

Reply:
because I use the object which used to delete in the line 99, so I have to reset the id

Reply:
you can consider upsert (ref: https://gorm.io/docs/create.html#upsert)